gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -84,11 +89,12 @@ class Node : public rclcpp::Node
   /**
    * \details This will call rclcpp::init if it hasn't been called yet.
    * \details Forwards arguments to the constructor for rclcpp::Node
+   * \param[in] qos Quality of service profile for the node entities.
    * \param[in] args List of arguments to pass to <a href="http://docs.ros2.org/latest/api/rclcpp/classrclcpp_1_1_node.html">rclcpp::Node</a>
    * \return A shared pointer to a new #gazebo_ros::Node
    */
   template<typename ... Args>
-  static SharedPtr CreateWithArgs(Args && ... args);
+  static SharedPtr CreateWithArgs(gazebo_ros::QoS & qos, Args && ... args);
```

Comment:
Is this parameter needed here?
I think that `static SharedPtr Get(sdf::ElementPtr _sdf);` can directly do all the work, without forwarding a parameter to this function.

Reply:
You're right. Removed the parameter in da5b314